### PR TITLE
Minor fixs to make torchbench runable on torch/xla

### PR DESCRIPTION
**Pull request by JackCaoG**
`import torch_xla.core.xla_model as xm` no longer trigger the xla runtime to init, hence explictly create the device here. This is a workaround for https://github.com/pytorch/xla/issues/4174.

`is_correct` reference has been deleted, I think it is a deadcode.

After this patch, I am able to run

```
python benchmarks/dynamo/torchbench.py --randomize-input --performance --trace-on-xla --training --backend=openxla --only resnet50
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @anijain2305

### CODE REVIEW

File: benchmarks/dynamo/common.py
Comment (wconstab) [inline benchmarks/dynamo/common.py:59]: what's the purpose of calling xla_device()?

i assume if it failed it would throw a different exception so catching importerror isn't enough, also its odd to bind it to a name.  is it enough to just do the import here and do the device call somewhere later at first use?

File: benchmarks/dynamo/common.py
Comment (wconstab) [inline benchmarks/dynamo/common.py:728]: why remove this? it is used by other backends, isn't it?

File: benchmarks/dynamo/common.py
Comment (JackCaoG) [reply benchmarks/dynamo/common.py:59]: the actual bug is discussed in https://github.com/pytorch/xla/issues/4174. I think it required pytorch/xla to registered something to the pytorch so backward can run correctly(when there is a cpu backward was run first). 

The reason that old workaround works is we used to eagerly init the runtime but now that doesn't happen until actual device is being init. I need to root cause this issue...

File: benchmarks/dynamo/common.py
Comment (JackCaoG) [reply benchmarks/dynamo/common.py:728]: `is_correct ` is actually not being init in this code. when I run it I got an error saying `is_correct ` is being used before assign.

`is_correct` is also not being used any places further down this function.  I suspect this is a dead code.

Comment (wconstab) [reply benchmarks/dynamo/common.py:728]: @anijain2305 do you know, are we still using 'speedup_experiment' for CI? do we not set 'should_check_outputs'?  maybe indeed this is dead and we should remove

File: benchmarks/dynamo/common.py
Comment (JackCaoG) [reply benchmarks/dynamo/common.py:59]: Init the device is not ideal as it actually make it impossible to run more than one tests at a same time on TPU, since the main process init the device outside of the `spawn` function and keep the device.

I tried a few other approach and they didn't work. I decided it is better to at least make training runable and fix this later in nightly

Comment (shunting314) [reply benchmarks/dynamo/common.py:59]: yea, let's make sure the script still works when XLA is missing.

Comment (JackCaoG) [reply benchmarks/dynamo/common.py:59]: it should, since the device init only happens when torch_xla import succed.

File: benchmarks/dynamo/common.py
Comment (shunting314) [reply benchmarks/dynamo/common.py:728]: `should_check_result` is set above if `args.randomize_input` is on. I think `randomize_input` is only used manually (not in CI).

File: benchmarks/dynamo/common.py
Comment (JackCaoG) [reply benchmarks/dynamo/common.py:59]: Let me try to dig into this issue a bit more this afternoon and see if I can workaround it.

File: benchmarks/dynamo/common.py
Comment (shunting314) [reply benchmarks/dynamo/common.py:728]: Let's remove the assignment to should_check_result as well.

My guess is maybe someone remove the definition of is_correct and the handling of is_correct some how. Removing this part looks good to me.

File: benchmarks/dynamo/common.py
Comment (shunting314) [reply benchmarks/dynamo/common.py:59]: I think a few things may make it better
1. have a API to explicitly initialize the xla runtime and call that rather than 'xla_device'. This can avoid some confusion.
2. or figure out the root cause :) 

But I'm ok to stamp the PR now to unblock. Unless @wconstab has other thoughts.

**Review by shunting314 — APPROVED** [review]

File: benchmarks/dynamo/common.py
Comment (JackCaoG) [reply benchmarks/dynamo/common.py:59]: Thanks Shunting, I added a API to explicitly init runtime. Let me update it.

File: benchmarks/dynamo/common.py
Comment (JackCaoG) [reply benchmarks/dynamo/common.py:728]: done

**Comment by JackCaoG** [comment]
```
This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0
- generated xml file: /var/lib/jenkins/workspace/test/test-reports/python-pytest/inductor.test_foreach/inductor.test_foreach-016b9206757a35ab.xml -
=========================== short test summary info ============================
FAILED [0.2053s] inductor/test_foreach.py::ForeachTests::test_2d_blocking__foreach_add - AssertionError: Scalars are not equal!

Expected 6 but got 5.
Absolute difference: 1
Relative difference: 0.16666666666666666
```

failure doesn't seem relevant

**Review by wconstab — APPROVED** [review]
agree that test_foreach is not relevant. if other tests all pass, you can merge with -f flag (or i can  help) but must wait for tests to finish before merging or -f will ignore still-pending tests

**Comment by JackCaoG** [comment]
Only test failure seems irrelevant. I will force merge

**Comment by JackCaoG** [comment]
PR was reverted because it touches XLA pin and there was another PR that also touches XLA pin got reverted. Reopen the pr and try to merge it again.

**Pull request closed without merging**
